build/tests/php/FileGenerator/PhelFnNormalizerTest.php

```diff
@@ -11,54 +11,161 @@
+
+        $expected = [
+            'fn-name-1' => [
+                [
+                    'fnName' => 'fn-name-1',
+                    'doc' => '',
+                    'fnSignature' => '',
+                    'desc' => '',
+                ],
+            ],
+            'fn-name-2' => [
+                [
+                    'fnName' => 'fn-name-2',
+                    'doc' => '',
+                    'fnSignature' => '',
+                    'desc' => '',
+                ],
+            ],
+        ];
+
+        self::assertSame($expected, $actual);
+    }
+
+    public function test_group_key_functions_in_same_group_with_question_mark(): void
+    {
+        $phelFnLoader = $this->createMock(PhelFnLoaderInterface::class);
+        $phelFnLoader->method('getNormalizedPhelFunctions')->willReturn([
+            'fn-name' => $this->createMock(PersistentMapInterface::class),
+            'fn-name?' => $this->createMock(PersistentMapInterface::class),
+        ]);
+
+        $normalizer = new PhelFnNormalizer($phelFnLoader);
+        $actual = $normalizer->getNormalizedGroupedPhelFns();
+
+        $expected = [
+            'fn-name' => [
+                [
+                    'fnName' => 'fn-name',
                     'doc' => '',
                     'fnSignature' => '',
                     'desc' => '',
                 ],
                 [
-                    'fnName' => 'test/table?',
+                    'fnName' => 'fn-name?',
                     'doc' => '',
                     'fnSignature' => '',
                     'desc' => '',
                 ],
+            ],
+        ];
+
+        self::assertSame($expected, $actual);
+    }
+
+
+    public function test_group_key_functions_in_same_group_with_minus(): void
+    {
+        $phelFnLoader = $this->createMock(PhelFnLoaderInterface::class);
+        $phelFnLoader->method('getNormalizedPhelFunctions')->willReturn([
+            'fn-name?' => $this->createMock(PersistentMapInterface::class),
+            'fn-name-' => $this->createMock(PersistentMapInterface::class),
+        ]);
+
+        $normalizer = new PhelFnNormalizer($phelFnLoader);
+        $actual = $normalizer->getNormalizedGroupedPhelFns();
+
+        $expected = [
+            'fn-name' => [
                 [
-                    'fnName' => 'test/table-',
+                    'fnName' => 'fn-name?',
                     'doc' => '',
                     'fnSignature' => '',
                     'desc' => '',
                 ],
                 [
-                    'fnName' => 'TEST/TABLE',
+                    'fnName' => 'fn-name-',
                     'doc' => '',
                     'fnSignature' => '',
                     'desc' => '',
                 ],
             ],
         ];
 
-        self::assertEquals($expected, $actual);
+        self::assertSame($expected, $actual);
+    }
+
+    public function test_group_key_functions_in_same_group_with_upper_case(): void
```

Comment:
This test and the previous makes sense to be more `ApiSearchGeneratorTest`, isn't it? 🤔

Reply:
Good question. But here you got both to compare:
<img width="1601" alt="Screenshot 2022-01-22 at 22 43 05" src="https://user-images.githubusercontent.com/5256287/150656423-c959a739-1596-4b61-8ecd-beace7e2e19c.png">
The key here is to understand the responsibilities of each of these classes and how are they connected.
- `PhelFnNormalizer`: normalize in an array the functions from phel (currently `$_GLOBAL`; and in the next version the Registry singleton). So in these tests `test_group_key_*` I want to verify that given those functions from phel, the normalizer will group them properly.
- `ApiSearchGenerator`: given the result of the `$groupNormalizedData` (which is the result from the `PhelFnNormalizer` behaviour) I want to generate a list of data-structures/arrays with the data I want to use for the search index.

As your eyes caught properly, these tests have a similar data in both somehow, but while in `PhelFnNormalizer` receive A and expect B, we use B in `ApiSearchGenerator` to verify C. Maybe now it's clearer?

Reply:
Yes, the output of one test in the input of the other 👌🏼